kratos/expression/variable_expression_data_io.cpp

```diff
@@ -62,8 +95,40 @@ VariableExpressionDataIO<TDataType>::VariableExpressionDataIO(const TDataType& S
     } else if constexpr(std::is_same_v<TDataType, Matrix>) {
         mShape.push_back(SampleValue.size1());
         mShape.push_back(SampleValue.size2());
-    } else { // this is for all array_1d types
+    } else if constexpr(
```

Comment:
The problem with templating in this case, if someone privdes an invalid template, it is a nightmare to find the actual error since the template error messages are too long to even read, and the candidates list will be too huge in this case. The current approach provides nice error message saying that there is a type mismatch, and see all the supported types at one glance.